aws_lambda_builders/workflows/nodejs_npm_esbuild/esbuild.py

```diff
@@ -102,3 +93,141 @@ def run(self, args, cwd=None):
             raise EsbuildExecutionError(message=err.decode("utf8").strip())
 
         return out.decode("utf8").strip()
+
+
+# The esbuild API flags are broken up into three forms (https://esbuild.github.io/api/):
+# Boolean types (--minify)
+SUPPORTED_ESBUILD_APIS_BOOLEAN = [
+    "minify",
+    "sourcemap",
+]
+
+# single value types (--target=es2020)
+SUPPORTED_ESBUILD_APIS_SINGLE_VALUE = [
+    "target",
+]
+
+# Multi-value types (--external:axios --external:aws-sdk)
```

Comment:
There is a fourth possibility to consider here - key-value types, e.g. the `--loader:x=y` option:

```
  --loader:X=L          Use loader L to load file extension X, where L is
                        one of: js | jsx | ts | tsx | css | json | text |
                        base64 | file | dataurl | binary
```

which I've tried to support in #343 as dictionary in the bundler_config

Reply:
I would imagine that this format still falls under the multi-value types as described in the esbuild API. Users will be able to configure loaders in their template by adding it in the metadata BuildProperties like
```
BuildProperties:
   Loader:
    - x=y
```
If you think this is can be improved, please let me know.

Reply:
I tried to make it more idiomatic by structuring it as a dictionary, as from what I understand, this option wouldn't permit multiple loaders for the same file extension e.g.:

```yaml
  BuildProperties:
    Loader:
      .ts: ts
      .grpc: text
```

Reply:
I completely agree that structuring it as a dict is likely more idiomatic. My concern is more around consistency of configuration to make it a better user experience. The more configuration methods we support, the more confusing it can get. I'd also like to mirror the esbuild API as closely as possible since I think that will make it easier to consume. I don't think we need to worry too much about validation as esbuild can validate its own APIs and return an appropriate error message.

Reply:
Fair enough - It will probably make it less confusing and easier to document the different types of options (as opposed to each option itself) i.e. negating the need to mirror the documentation of esbuild too closely.